### ISSUE TITLE
AggregateBalance reserved model name

### ISSUE THREAD

**Issue by cyrus-za** [prisma/prisma#3220]
Why is `AggregateBalance` a reserved name for models?

<img width="451" alt="image" src="https://user-images.githubusercontent.com/1845861/89504902-117ca080-d7c9-11ea-81f7-9213d0449dc0.png">

**Comment by WellFail** [comment]
@cyrus-za 
Because if you create a model named `Balance`, the` prisma client` will try to generate an AggregateBalance type for `aggregateApi` which will conflict with `AggregateBalance` model.

**Comment by WellFail** [comment]
https://github.com/prisma/prisma/blob/c36c1a7a84a5be3fd12522870a2f48cabdb6b833/src/packages/client/src/generation/generateClient.ts
https://github.com/prisma/prisma-engines/blob/b1c5396ac9832eb8481b3c7f36c393ab1747211d/libs/datamodel/core/src/transform/ast_to_dml/reserved_model_names.rs

There you can look at all deny model name list

**Comment by cyrus-za** [comment]
Thanks. Just a little strange

**Comment by janpio** [comment]
Valid concern and we will see if we can find a way around this in some way.

**Comment by WellFail** [comment]
@janpio 
https://github.com/prisma/prisma/blob/c36c1a7a84a5be3fd12522870a2f48cabdb6b833/src/packages/client/src/generation/generateClient.ts

`Aggregate` currently on this list and I see Aggregate${modelName} type in my generated prisma client.

**Comment by pantharshit00** [comment]
Yes, I can confirm this and the writeup @WellFail did is correct. We will need to prefix them as did for `${name}Client`.
